### PR TITLE
Add danger zone to circle admin page

### PR DESCRIPTION
**Pull request by sshmm**
## Motivation and Context
implements #966

## Description

1- Add deleted_at column to circles table
2- Add deleteCircle action
3- add danger zone to circle admin page
4- add remove circle modal with remove and cancel buttons
5- on remove circle the circle will be removed and navigation to overview page happens
6- on error show snackbar error

## Test and Deployment Plan

precondition : circle admin
1- Go to circle Admin page and click remove circle button in danger zone
2a - click remove circle button in remove circle modal
Expected:
Navigation to overview page and the circle should not appear from it
2b- click cancel button
Expected:
the modal should be closed
2c- hardcode a wrong circle number and click remove
Expected:
Snackbar should appear with error

## Screenshots (if appropriate):
image

### CODE REVIEW

**Comment by crabsinger** [comment]
@sshmm seems this PR get mixed with lots of other commits - so its hard to review and and approve.

Can you take this commit https://github.com/coordinape/coordinape/commit/fd5825bb486bbdc02e042f06b354a9ca9056c85d

and have a separate PR that is just the hasura deleteCircle action and hasura schema [api/hasura/actions/_handlers/deleteCircle.ts](https://github.com/coordinape/coordinape/commit/fd5825bb486bbdc02e042f06b354a9ca9056c85d#diff-1b73b72aeab67903e6de1ea2e5c440deb7d54a357edb70a6900fc139b9076698)

without any frontend changes associated. Then we can merge that first, and separately merge the frontend changes adn everything else afterwards. Feel free to discuss how to break this thread up in Discord as well.

In short, break this into the smallest PRs that we can.

**Comment by crabsinger** [comment]
I merged main into the base branch, but still seems the diff here is huge... not sure why or if GH is being slow.

**Comment by sshmm** [comment]
> I merged main into the base branch, but still seems the diff here is huge... not sure why or if GH is being slow.

I will create a new branch on of 1.5.6 and cherry pick my changes later tomorrow(today my time) then update the PR

**Comment by sshmm** [comment]
@crabsinger I merged 1.5.6 to my branch and now you can see my changes only.

**Comment by crabsinger** [comment]
@sshmm ok - should I update the 1.5.6 branch again from main?

What are next steps or questions for moving this forwards?

**Comment by sshmm** [comment]
> 

Yes please so that I will be able to update this PR

**Comment by sshmm** [comment]
replaced by #1262
